apis/nucleus/src/components/listbox/ListBox.jsx

```diff
@@ -160,7 +162,13 @@ export default function ListBox({
                 const processedPages = postProcessPages ? postProcessPages(p) : p;
                 local.current.validPages = true;
                 listData.current.pages = processedPages;
-                setPages(processedPages);
+
+                if (currentPage === 1) {
+                  setPages(processedPages);
+                } else {
+                  setPages((_pages) => [...(_pages || []), ...processedPages]);
+                }
```

Comment:
I'm not really sure what you are trying to do here

Reply:
there is another issue with list about not rendering all items in list 
imagine if we selected 207 items 
it will show only 7 items in list not the last 107 items (100 from previous, 7 from current api call)

by doing so, we are tracking the page and if we got reseted it (which has been detected by `qTop` value when it becomes 1) then we know that we should start the list from scratch and remove old items from list (`setPages(processedPages)` part) 

if we are incrementing the page (which means going down in list) then we might keep adding items to the `pages` state in order to preserve them and prevent half-rendered list issue

Reply:
This is what it fixes!
as you can see Items 440 in list is missing!

IN new commit i modified this part as well, we dont need to calculate the `currentPage` and the resetting happens in `fetchData` function.
![Screenshot 2022-07-22 at 08 04 13](https://user-images.githubusercontent.com/29652890/180382004-ed4be1ce-8741-49cb-9725-246e0b511bab.png)